walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java

```diff
@@ -22,6 +22,6 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
-    return "Hello " + name;
+    return "Hello " + name.replaceAll("[^a-zA-Z0-9\\s+]", "").trim();
```

Comment:
```suggestion
    return "Hello " + name.replaceAll("[\$@%#]", "").trim();
```
Oops so this is going to replace all lowercase and uppercase alphabet characters along with numbers and spaces with ''. Instead what you want to put inside the brackets are the special characters like so: \$@%#
Make sure the $ is escaped as that is a special regex symbol and would otherwise not read the $. Regex is a way to match/find patterns in text. Typically it's 1-1 so if you have 'abc' it looks for the first reference of abc in your text. However the brackets [ ] allow you to be less strict so that you don't have to exactly know/care about how the pattern is, just that it follows some sort of guideline i.e. [0-9] matches the first number regardless of what it is.
In case you didn't, I'd recommend running your tests before and after the changes to verify what you are doing is correct.

Reply:
Ahh this makes sense!! I changed it to [$@%#]. I tried to escape the \$ and it was threw an illegal escape character, so I removed the \ and it passed the test.